### ISSUE TITLE
able to crash sails when querying mongo

### ISSUE THREAD

**Issue by cortfritz** [balderdashy/sails#2202]
I am saving telemetry in mongo.  In a (accidentally malformed) blueprint query for a particular set of telemetry entries, I crashed sails - apparently due to a bug between sails and mongo.

#### blueprint url 
Here is the blueprint url I was using against the model ````Journal````.  "````/bp````" is the blueprint prefix in sails.config.blueprint.

````
/bp/journal?type=telemetry&data.logFormat:0.0.6
````

That url was a mistake.  I had meant to enter this:
````
/bp/journal?type=telemetry&data.logFormat=0.0.6
````
...substituting the correct "=" for the incorrect and JSONy ":".  

Also, "logFormat" is a property of the JSON object "data".  Not sure if this is the right way to query.

#### error
````
/Users/cfritz/Documents/apps/mkultra/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/connection/base.js:245
        throw message;      
              ^
TypeError: Object [a lifted Sails app on port 1337] has no method 'error'
    at journalCreatedCallback (/Users/cfritz/Documents/apps/mkultra/api/services/JournalService.js:60:19)
    at bound (/Users/cfritz/Documents/apps/mkultra/node_modules/lodash/dist/lodash.js:957:21)
    at applyInOriginalCtx (/Users/cfritz/Documents/apps/mkultra/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:416:80)
    at wrappedCallback (/Users/cfritz/Documents/apps/mkultra/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:326:16)
    at _normalizeCallback.callback.error (/Users/cfritz/Documents/apps/mkultra/node_modules/sails/node_modules/waterline/node_modules/node-switchback/lib/normalize.js:42:31)
    at _switch (/Users/cfritz/Documents/apps/mkultra/node_modules/sails/node_modules/waterline/node_modules/node-switchback/lib/factory.js:33:28)
    at /Users/cfritz/Documents/apps/mkultra/node_modules/sails/node_modules/waterline/lib/waterline/query/dql/create.js:194:14
    at bound (/Users/cfritz/Documents/apps/mkultra/node_modules/lodash/dist/lodash.js:957:21)
    at applyInOriginalCtx (/Users/cfritz/Documents/apps/mkultra/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:416:80)
    at wrappedCallback (/Users/cfritz/Documents/apps/mkultra/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:326:16)
````

**Comment by sailsbot** [comment]
Thanks for posting, @cortfritz.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
